src/GraphQL/Types/ArgumentInformation.cs

```diff
@@ -0,0 +1,179 @@
+        /// <br/><br/>
+        /// If an expression is returned, it will be used to populate the method argument within the field resolver;
+        /// if not, the query argument's value will be used to populate the method argument within the field resolver.
+        /// <br/><br/>
+        /// The default implementation will return either a <see cref="QueryArgument"/> or <see cref="LambdaExpression"/>
+        /// instance; not both. It is possible to return both, in which case the query argument will be added to the
+        /// field and the expression will be used to populate the method argument within the field resolver.
+        /// You cannot return <see langword="null"/> for both the query argument and expression.
+        /// </summary>
+        public virtual (QueryArgument? QueryArgument, LambdaExpression? Expression) ConstructQueryArgument()
+        {
+            if (Expression != null)
+                return (null, Expression);
+
+            var type = TypeInformation.ConstructGraphType();
+            var argument = new QueryArgument(type)
+            {
+                Name = ParameterInfo.Name!,
+                Description = ParameterInfo.Description(),
+                DefaultValue = ParameterInfo.IsOptional ? ParameterInfo.DefaultValue : null,
+            };
+            return (argument, null);
+        }
+    }
+
+    /// <inheritdoc/>
+    public class ArgumentInformation<TReturnType> : ArgumentInformation
+    {
+        /// <inheritdoc cref="ArgumentInformation.ArgumentInformation(ParameterInfo, Type, FieldType, TypeInformation, LambdaExpression?)"/>
+        public ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, FieldType fieldType, TypeInformation typeInformation, Expression<Func<IResolveFieldContext, TReturnType>>? expression)
+            : base(ValidateParameterInfo(parameterInfo), sourceType, fieldType, typeInformation, expression)
+        {
+        }
+
+        /// <inheritdoc/>
+        public ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, FieldType fieldType, TypeInformation typeInformation)
+            : base(ValidateParameterInfo(parameterInfo), sourceType, fieldType, typeInformation)
+        {
+        }
+
+        /// <summary>
+        /// Validates that the <see cref="ParameterInfo"/> supplied to the constructor has a return type
+        /// that matches the <typeparamref name="TReturnType"/> of this instance.
+        /// </summary>
+        private static ParameterInfo ValidateParameterInfo(ParameterInfo parameterInfo)
+        {
+            if (parameterInfo.ParameterType != typeof(TReturnType))
+            {
+                throw new ArgumentOutOfRangeException(nameof(parameterInfo), $"Parameter must have a return type of {typeof(TReturnType).Name}.");
+            }
+            return parameterInfo;
+        }
+
+        /// <summary>
+        /// Gets or sets a delegate in the form of a <see cref="LambdaExpression"/> to be used to populate
```

Comment:
> in the form of a \<see cref="LambdaExpression"/>

copy-pasted

Reply:
Yeah, still, it's accurate.  I can add an additional note.

Reply:
I'm going to rename `TReturnType` to `TArgumentType` also to better describe it.

Reply:
or maybe `TParameterType`

Reply:
Changed.